### PR TITLE
add new Hive reader

### PR DESCRIPTION
**Pull request by dengkeshun**
# Description

add a new Hive reader.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] New Loader/Tool
- [ ] Bug fix / Smaller change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have added a library.json file if a new loader/tool was added
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by EmanuelCampos — COMMENTED** [review]
nice one, the CI is broken can you make sure to run `make format; make lint`?

**Comment by dengkeshun** [comment]
make format; make lint have been done.

File: llama_hub/hive/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    auth='NOSASL'
+)
+
+query = "SELECT * FROM p1 LIMIT 100"
+documents = reader.load_data(
+    query=query
+)
+```
+
+This loader is designed to be used as a way to load data into [LlamaIndex](https://github.com/run-llama/llama_index/tree/main/llama_index) and/or subsequently used as a Tool in a [LangChain](https://github.com/hwchase17/langchain) Agent. See [here](https://github.com/emptycrown/llama-hub/tree/main) for examples.
```
Comment (anoopshrma) [inline llama_hub/hive/README.md:28]: Could you change the llama-hub link to this: https://github.com/run-llama/llama-hub/tree/main/llama_hub

File: llama_hub/hive/base.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        database: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        auth: Optional[str] = None,
+    ):
+        """Initialize with parameters."""
+        try:
+            from pyhive import hive
+        except ImportError:
+            raise ImportError("`hive` package not found, please run `pip install hive`")
```
Comment (anoopshrma) [inline llama_hub/hive/base.py:38]: It should be `pip install pyHive` 

https://pypi.org/project/PyHive/

**Review by anoopshrma — APPROVED** [review]
Everything looks good! We are good for merge if linting gets fixed!!

**Comment by dengkeshun** [comment]
need to run make format; make lint for any changes . Unn, this step missing.

**Pull request merged**